### PR TITLE
YLP-1697 Split patients and team members management

### PR DESCRIPTION
**Pull request by TimCanu**

### CODE REVIEW

**Comment by TimCanu** [comment]
- A test is currently skipped in patient/hook.test.tsx and should be investigated (if no solution found, I advised to delete the failing assert)
- Some more manual testing is required
- lib/patient/utils could be a bit more unit tested
- E2E tests passed in a previous commit but need to be relaunched (I advise creating a new beta and check the CI)

File: packages/yourloops/layout/main-layout.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/**
```
Comment (fatb38) [inline packages/yourloops/layout/main-layout.tsx:1]: There is already a layout folder in components folder, maybe we can move `dashboard-layout` and `menu-layout` in this new folder.

File: packages/yourloops/lib/patient/patient-api.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  static async updatePatientAlerts(teamId: string, patientId: string, monitoring: Monitoring): Promise<void> {
+    await HttpService.put<void, Monitoring>({
+      url: `/crew/v0/teams/${teamId}/patients/${patientId}/monitoring`,
+      payload: monitoring
+    })
+  }
+
+  static async removePatient(teamId: string, userId: string): Promise<void> {
+    await HttpService.delete({ url: `/crew/v0/teams/${teamId}/patients/${userId}` })
```
Comment (fatb38) [inline packages/yourloops/lib/patient/patient-api.ts:79]:
```suggestion
  static async removePatient(teamId: string, patientId: string): Promise<void> {
    await HttpService.delete({ url: `/crew/v0/teams/${teamId}/patients/${patientId}` })
```

File: packages/yourloops/test/unit/lib/patient/team-api.test.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import { ITeamMember } from '../../../../models/team'
+import { AxiosResponse } from 'axios'
+import { INotificationAPI } from '../../../../models/notification'
+import { UserRoles } from '../../../../models/user'
+import { HttpHeaderKeys } from '../../../../models/api'
+import { getCurrentLang } from '../../../../lib/language'
+import PatientApi from '../../../../lib/patient/patient-api'
+
+describe('PatientApi', () => {
+  const userId = 'userId'
```
Comment (fatb38) [inline packages/yourloops/test/unit/lib/patient/team-api.test.tsx:38]:
```suggestion
  const patientId = 'patientId'
```

**Review by fatb38 — CHANGES_REQUESTED** [review]

File: packages/yourloops/lib/patient/hook.tsx [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+import { errorTextFromException } from '../utils'
+import { PatientContextResult } from './provider'
+
+export default function usePatientProviderCustomHook(): PatientContextResult {
+  const { cancel: cancelInvitation, getInvitation } = useNotification()
+  const { removeTeamFromList } = useTeam()
+  const { user, getFlagPatients, flagPatient } = useAuth()
+
+  const [patients, setPatients] = useState<Patient[]>([])
+  const [initialized, setInitialized] = useState<boolean>(false)
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/lib/patient/hook.tsx:52]: To be discussed as I am not sure to fully understand

File: packages/yourloops/lib/patient/hook.tsx [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    const invitation = notificationConversion(apiInvitation)
+    const patientTeam = {
+      status: UserInvitationStatus.pending,
+      teamId: team.id,
+      remoteMonitoringEnabled: team.monitoring?.enabled
+    }
+
+    let patient: Patient = patients.find(patient => patient.profile.email === invitation.email)
+    if (!patient) {
+      patient = {
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/lib/patient/hook.tsx:113]: Is it just to make it prettier or it is duplicated code?

File: packages/yourloops/lib/patient/patient-api.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  teamId: string
+  email: string
+}
+
+interface InvitePatientPayload extends InvitePatientArgs {
+  role: UserRoles
+}
+
+export default class PatientApi {
+  static async getPatients(): Promise<ITeamMember[]> {
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/lib/patient/patient-api.ts:48]: To be discussed as I am not sure what you mean

File: packages/yourloops/lib/patient/utils.tsx [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+  static isOnlyPendingInvitation = (patient: Patient): boolean => {
+    const tm = patient.teams.find((team: PatientTeam) => team.status !== UserInvitationStatus.pending)
+    return typeof tm === 'undefined'
+  }
+
+  static isInTeam = (patient: Patient, teamId: string): boolean => {
+    const tm = patient.teams.find((team: PatientTeam) => team.teamId === teamId)
+    return typeof tm === 'object'
+  }
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/lib/patient/utils.tsx:103]: Replacing `return typeof tm === 'undefined'` with `return !tm` does not feel right, are you sure?

File: packages/yourloops/lib/patient/patient-api.ts (hunk 2 above)
Comment (TimCanu) [reply packages/yourloops/lib/patient/patient-api.ts:79]: Patients have a userId. This one makes sense

**Review by fatb38 — APPROVED** [review]

**Pull request merged**